server/service/client.go

```diff
@@ -1062,11 +1065,19 @@ func (c *Client) doGitOpsPolicies(config *spec.GitOps, logFn func(format string,
 		numPolicies := len(config.Policies)
 		logFn("[+] syncing %d policies\n", numPolicies)
 		if !dryRun {
-			// Note: We are reusing the spec flow here for adding/updating policies, instead of creating a new flow for GitOps.
-			if err := c.ApplyPolicies(config.Policies); err != nil {
-				return fmt.Errorf("error applying policies: %w", err)
+			totalApplied := 0
+			for i := 0; i < len(config.Policies); i += batchSize {
+				end := i + batchSize
+				if end > len(config.Policies) {
+					end = len(config.Policies)
+				}
+				totalApplied += end - i
```

Comment:
I made a few attempts to abstract the batching logic across the changes using generics, but it felt a bit too clever.  Opted for some code duplication, but open to suggestions.

Reply:
I'm ok with simple code, but you could always create a function `applyBatch(func, ...)` that receives the func to call in batch.